compiler/noirc_evaluator/src/ssa/function_builder/mod.rs

```diff
@@ -556,10 +556,10 @@ impl std::ops::Index<BasicBlockId> for FunctionBuilder {
 fn validate_numeric_type(typ: &NumericType) {
     match &typ {
         NumericType::Signed { bit_size } => match bit_size {
-            8 | 16 | 32 | 64 | 128 => (),
+            1 | 8 | 16 | 32 | 64 | 128 => (),
```

Comment:
Why are we allowing `i1` now?
The type would just be a single sign bit so I'm not sure it makes much sense. I saw your previous comment about the inconsistent result when printing zero versus comparing with zero but maybe we could add special logic to casting to avoid the `i1` intermediate?

Reply:
We ban it in the frontend https://github.com/noir-lang/noir/pull/8172 so we should ban it in SSA as well. Not sure why it is being used, if it is we should move away from it.

Reply:
In the end a truncate was missing.